### PR TITLE
Intake subsystem

### PR DESCRIPTION
**Pull request by Basil070104**

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -40,6 +40,15 @@
         public static final double INTAKE_FLYWHEELS_FORWARD_POWER = 0.3;
         public static final double INTAKE_FLYWHEELS_REVERSE_POWER = -0.3;
         public static final int INTAKE_STOP = 0;
+        public static final int PCM_ID = 3;
+        public static final int FORWARD_CHANNEL = 2;
+        public static final int REVERSE_CHANNEL = 3;
+        public static final int FIRST_MOTOR_CONTROLLER_ID = 10;
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/Constants.java]: change to MASTER_ID

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -40,6 +40,15 @@
...
+        public static final int SECOND_MOTOR_CONTROLLER_ID = 11;
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/Constants.java]: change to FOLLOWER_ID

File: src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -40,6 +40,15 @@
...
+        public static final int INWARD = -1
+        public static final int OUTWARD = 1
+        public static final int STOP = 0;
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/Constants.java]: you dont need a speed for stop

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 4]
```diff
@@ -10,32 +11,36 @@
 
     public static final WPI_VictorSPX intakeMotor = new WPI_VictorSPX(IntakeConstants.INTAKE_WHEELS_ID);
     public static final WPI_VictorSPX intakeBarMotor = new WPI_VictorSPX(IntakeConstants.INTAKE_BAR_MOTOR_ID);
+    public DoubleSolenoid test = new DoubleSoleniod(IntakeConstants.PCM_ID, IntakeConstants.FORWARD_CHANNEL, IntakeConstants.REVERSE_CHANNEL);
+    public IntakeMotor = new IntakeMotor(IntakeConstants.FIRST_MOTOR_CONTROLLER_ID, IntakeMotor.SECOND_MOTOR_CONTROLLER_ID, IntakeMotor.INWARD, IntakeMotor.OUTWARD, IntakeMotor.STOP);
+    public DigitalInput input = new DigitalInput(0);
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/subsystems/Intake.java]: change input to pos0

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 5]
```diff
@@ -10,32 +11,36 @@
 
     public static final WPI_VictorSPX intakeMotor = new WPI_VictorSPX(IntakeConstants.INTAKE_WHEELS_ID);
     public static final WPI_VictorSPX intakeBarMotor = new WPI_VictorSPX(IntakeConstants.INTAKE_BAR_MOTOR_ID);
+    public DoubleSolenoid test = new DoubleSoleniod(IntakeConstants.PCM_ID, IntakeConstants.FORWARD_CHANNEL, IntakeConstants.REVERSE_CHANNEL);
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/subsystems/Intake.java]: this does not need to be named test

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 6]
```diff
@@ -10,32 +11,36 @@
...
+    public IntakeMotor = new IntakeMotor(IntakeConstants.FIRST_MOTOR_CONTROLLER_ID, IntakeMotor.SECOND_MOTOR_CONTROLLER_ID, IntakeMotor.INWARD, IntakeMotor.OUTWARD, IntakeMotor.STOP);
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/subsystems/Intake.java]: its should be 2 motors, and you instantiate them as WPI_VictorSPX

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 7]
```diff
@@ -10,32 +11,36 @@
...
     }
 
     public void retract() {
-        intakeBarMotor.set(ControlMode.PercentOutput, -(IntakeConstants.INTAKE_BAR_MOTOR_SPEED));
+        test.set(DoubleSolenoid.Value.kreverse);
     }
 
     public void getPosition() {
-        
+        input.get(DigitalInput);
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/subsystems/Intake.java]: remove DigitiInput from within the brackets

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 8]
```diff
@@ -10,32 +11,36 @@
...
     }
 
     public void spinIn(){
-        intakeMotor.set(ControlMode.PercentOutput, IntakeConstants.INTAKE_FLYWHEELS_FORWARD_POWER);
+        test.set(IntakeMotor.Value.inward);
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/subsystems/Intake.java]: you set motor speed like, motor_name.set(ControlMode.PercentOutput, NAME OF YOUR CONSTANT)

**Review by MichaelCrandall — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (Basil070104) [reply src/main/java/frc/robot/Constants.java]: Line 45?

File: src/main/java/frc/robot/Constants.java (hunk 2 above)
Comment (Basil070104) [reply src/main/java/frc/robot/Constants.java]: Line 47?

File: src/main/java/frc/robot/subsystems/Intake.java (hunk 6 above)
Comment (Basil070104) [reply src/main/java/frc/robot/subsystems/Intake.java]: Like the ones at the top

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 9]
```diff
@@ -8,12 +8,10 @@
...
 public class Intake extends SubsystemBase {
 
-
-    public static final WPI_VictorSPX intakeMotor = new WPI_VictorSPX(IntakeConstants.INTAKE_WHEELS_ID);
-    public static final WPI_VictorSPX intakeBarMotor = new WPI_VictorSPX(IntakeConstants.INTAKE_BAR_MOTOR_ID);
-    public DoubleSolenoid test = new DoubleSoleniod(IntakeConstants.PCM_ID, IntakeConstants.FORWARD_CHANNEL, IntakeConstants.REVERSE_CHANNEL);
-    public IntakeMotor = new IntakeMotor(IntakeConstants.FIRST_MOTOR_CONTROLLER_ID, IntakeMotor.SECOND_MOTOR_CONTROLLER_ID, IntakeMotor.INWARD, IntakeMotor.OUTWARD, IntakeMotor.STOP);
-    public DigitalInput input = new DigitalInput(0);
+    public DoubleSolenoid = new DoubleSoleniod(IntakeConstants.PCM_ID, IntakeConstants.FORWARD_CHANNEL, IntakeConstants.REVERSE_CHANNEL);
+    public static final WPI_VictorSPX intakeMotorOne = new WPI_VictorSPX(IntakeConstants.MASTER_ID);
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/subsystems/Intake.java]: remove static final

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 10]
```diff
@@ -8,12 +8,10 @@
...
+    public static final WPI_VictorSPX intakeMotorTwo = new WPI_VictorSPX(IntakeConstants.FOLLOWER_ID);
+    public DigitalInput input = new DigitalInput(getPosition(0););
 
 
     public Intake(){
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/subsystems/Intake.java]: we are going to add some fancy code here, go to the link i gave you about victor motors and look for .configFactoryDefault(), .follow(), and .setInverted()

**Review by MichaelCrandall — CHANGES_REQUESTED** [review]

**Pull request merged**